### DISCUSSION TITLE
UI 컴포넌트 관련 추상화 컨벤션에 대한 필요성

### DISCUSSION THREAD

**Discussion by godhyeongman** [eRentronic/eRentronic-Client#58] [아이디어]
## 주제
UI 컴포넌트에서 추상화 과정이 필요한지 아닌지에 대한 건의

## 상황
template 에서는 데이터를 다루는 과정이 중요하여 UI 로직을 세부 추상화 된 컴포넌트 에서 내부적으로 처리함

UI 컴포넌트 단에서는 UI다루는 로직만이 존재하고 이가 주 관심사여서 가장 최상단부에 이에 대한 로직과 상태가 존재하는데 이를 JSX 구문을 추상화 하는것이 의미 있는가에 대한 의견 필요

## 현재 효과

template 단에서는 필요한 부분만 추상화 하는것이 가독성이 좋아짐을 발견함

UI 컴포넌트 단에서는 JSX 구현부의 레이아웃 정도를 판단하는데 도움이 되는 효과뿐(세부 마크업만 분담함) 오히려 변경에 분리할 수 있다고도 판단됨

**Comment by herrakam** [comment]
계속 이 부분에 대해서 생각을 해 봤는데, 검증 로직을 제대로 구현하지 않고 조건부 랜더링으로 검증 로직을 해결하려는게  문제가 될 수 있을거 같습니다. 그렇다고 기존의 방법인 조건부 렌더링을 버릴 필요는 없는 것 같아요.  사용자 편의성 방면으로는 장점이 있는 방법 같아서요.  현재 jsx 내의 조건부 랜더링은 유지를 하되, 검증 함수를 추가로 구현하는건 어떤가요?  조건부 랜더링이 부담하고 있는 검증 부분을 분리하면서 사용자 편의성도 챙길 수 있을 것 같아용

**Reply by godhyeongman** [reply]
엇! 뭔가 서로 다른 방향으로 문제를 받아들인 것 같네요! 😗
검증 로직 과 조건부 렌더링이 언급된 이유를 알 수 있을까요?

추가로 질문 의도는 '데이터를 처리하는 컴포넌트에서 UI 처리 로직을 추상화' 로 숨기는건 의미가 있다 생각을 하는데 'UI를 처리하는 컴포넌트에서 컴포넌트를 나누어서 단순 세부 마크업만 숨기는 것이 의미 있는 것인가?' 라는게 궁금했습니다!

**Reply by herrakam** [reply]
아 다른 문제를 말씀하시는 거였군요 ㅋㅋㅋㅋㅋㅋㅋㅋ  전 데이터 다루는 컴포넌트만큼은 아니겠지만 ui 처리하는 컴포넌트들도 나눌 가치가 있다고 생각해요! 일단 너무 길면 가독성이 문제가 발생할거고, 데이터를 다루는 컴포넌트들만 나누게 되면 ui 처리 컴포넌트와 데이터 처리 컴포넌트를 둘 다 가지고 있는 템플릿 이상의 컴포넌트들에서는 불균형 적으로 보일 것 같습니다!!

**Reply by godhyeongman** [reply]
동의합니다! 이부분은 결정한걸로 진행하는게 좋을것 같습니다 😃
